keysign/util.py

```diff
@@ -363,17 +365,22 @@ def get_local_bt_address(hci_number=0):
 
 def is_bt_available(hci_number=0):
     """We check for bluez with a deferred thread because this is a blocking operation"""
-    d = threads.deferToThread(_get_bluez, hci_number)
+    d = threads.deferToThread(get_bt_power, hci_number)
     return d
 
 
-def _get_bluez(hci_number=0):
-    """If the bluez object is available it means that there is a working Bluetooth"""
+def get_bt_power(hci_number=0):
+    """Check if there is a powered on Bluetooth device"""
     bus = dbus.SystemBus()
     try:
-        bus.get_object("org.bluez", "/org/bluez/hci%i" % hci_number)
-        log.debug("Bluetooth seems to be available in the system")
-        return True
+        adapter = dbus.Interface(bus.get_object("org.bluez", "/org/bluez/hci%i" % hci_number),
+                                 "org.freedesktop.DBus.Properties")
+        power = adapter.Get("org.bluez.Adapter1", "Powered")
+        if power:
+            log.debug("Bluetooth seems to be available in the system")
+        else:
+            log.debug("Bluetooth seems to be turned off")
+        return power
```

Comment:
I don't like having too many places where we return from a function. I'd appreciate a single return at the end, but that shouldn't block merging this.

With this structure we cannot distinguish no adapter from no-power.
Do you think we can raise something like `NoAdapter(BluetoothException)` and `UnpoweredAdapter(BluetoothException)` or so?
Then we can catch those and eventually show a widget with a button to have Bluetooth turned on.
Again, that shouldn't prevent us from merging the current state, but I think we'd be happier in the long run if we were able to discriminate the several possible failures.